l2-contracts/contracts/AttesterRegistry.sol

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+contract AttesterRegistry {
+    // Owner of the contract (the ConsensusAuthority contract).
+    address public owner;
+    // A map of node owners => attesters (used for attester lookups).
+    mapping(address => Attester) public attesters;
+    // An array to keep track of attester node owners (used for iterating attesters).
+    address[] public attesterOwners;
+    // The current committee list. Weight and public key are stored explicitly
+    // since they might change after committee selection.
+    CommitteeAttester[] public committee;
+
+    struct Attester {
+        // Voting weight.
+        uint256 weight;
+        // ECDSA public key.
+        bytes pubKey;
+        // A flag stating if the attester is inactive. Inactive attesters are not
+        // considered when selecting committees. Only inactive attesters can
+        // be removed from the registry.
+        bool isInactive;
```

Comment:
Seems like currently this doesn't really have any effect, we just need to mark the attester as inactive and then we can delete it. Is there any value in this extra step? Do we plan to add some cooldown in the future? Shall we maybe then also add this field in the future, when we actually need it?

Reply:
Nodes can mark themselves as inactive and they are not selected in the next committee. But only governance can call remove method.